File: apps/api/src/app/exerciseMuscle.entity.ts
import { Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { ExercisesEntity } from './exercises/exercises.entity';
import { MuscleEntity } from './muscle.entity';

@Entity({ name: 'exercise_muscle', schema: 'public' })
export class ExerciseMuscleEntity {
  @PrimaryGeneratedColumn('increment')
  id: number;

  @ManyToOne(() => ExercisesEntity, (exercise) => exercise.id, {
    cascade: true,
    eager: true,
  })
  @JoinColumn({ name: 'exercise_id' })
  exercise: ExercisesEntity;

  @ManyToOne(() => MuscleEntity, (muscle) => muscle.id, {
    cascade: true,
    eager: true,
  })
  @JoinColumn({ name: 'muscle_id' })
  muscle: MuscleEntity;
}

File: my-workout-api/src/main/java/com/avernet/my_workout_api/ExerciseMuscle/ExerciseMuscleEntity.java
package com.avernet.my_workout_api.ExerciseMuscle;

import com.avernet.my_workout_api.Muscle.MuscleEntity;
import com.avernet.my_workout_api.exercise.ExerciseEntity;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "muscle_group")
@Data
public class ExerciseMuscleEntity {
    @Id
    Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    ExerciseEntity exercise;

    @ManyToOne(fetch = FetchType.LAZY)
    MuscleEntity muscle;
}
